catalog/ui/search-ui/search-endpoint/src/test/java/org/codice/ddf/ui/searchui/query/controller/SearchControllerTest.java

```diff
@@ -108,6 +136,37 @@ public void setUp() throws Exception {
         mockServerSession = mock(ServerSession.class);
 
         when(mockServerSession.getId()).thenReturn(MOCK_SESSION_ID);
+
+        futureMap = new HashMap<>();
+        futureMap.put("key", mockFuture);
+
+        when(mockSearchRequest.getId()).thenReturn("myTestId");
+        when(mockSearchRequest.getQuery()).thenReturn(mockQuery);
+        when(mockQuery.getSortBy()).thenReturn(SortBy.NATURAL_ORDER);
+    }
+
+    @Test
+    public void testExceptionPassedOnInterrupt() throws Exception {
```

Comment:
@lessarderic This test (and similar ones found below) are the reason for the factory methods. Let me know if you find an alternative.